src/main/src/main/java/org/geoserver/catalog/impl/CascadedLayerInfoImpl.java

```diff
@@ -0,0 +1,254 @@
+
+    public CascadedLayerInfoImpl() {}
+
+    public CascadedLayerInfoImpl(LayerInfo delegate) {
+
+        super.setResource(delegate.getResource());
+        super.setName(delegate.getResource().getName());
+        super.setEnabled(delegate.enabled());
+        super.setType(delegate.getType());
+    }
+
+    @Override
+    public void reset() {
+        // select all formats for use
+        selectedRemoteStyles.addAll(getRemoteStyles());
+        // set first style is selected
+        if (!selectedRemoteStyles.isEmpty()) forcedRemoteStyle = selectedRemoteStyles.get(0);
+        // select all styles available for us
+        selectedRemoteFormats.addAll(getAvailableFormats());
+    }
+
+    @Override
+    public List<String> getRemoteStyles() {
+
+        WMSLayerInfo layerInfo = (WMSLayerInfo) super.resource;
+
+        try {
+            // read from cap doc
+            return layerInfo
+                    .getWMSLayer(null)
+                    .getStyles()
+                    .stream()
+                    .map(s -> s.getName())
+                    .collect(Collectors.toList());
+
+        } catch (Exception e) {
+            LOGGER.log(
+                    Level.SEVERE,
+                    "Unable to fetch styles for cascaded layer " + layerInfo.getName());
+            LOGGER.log(Level.SEVERE, e.getMessage());
+        }
+
+        return Collections.EMPTY_LIST;
+    }
+
+    @Override
+    public String getForcedRemoteStyle() {
+        return forcedRemoteStyle;
+    }
+
+    @Override
+    public void setForcedRemoteStyle(String forcedRemoteStyle) {
+        this.forcedRemoteStyle = forcedRemoteStyle;
+    }
+
+    @Override
+    public List<String> getAvailableFormats() {
+        WMSLayerInfo layerInfo = (WMSLayerInfo) super.resource;
+        try {
+            return layerInfo
+                    .getStore()
+                    .getWebMapServer(null)
+                    .getCapabilities()
+                    .getRequest()
+                    .getGetMap()
+                    .getFormats()
+                    .stream()
+                    .filter(CascadedLayerInfoImpl::isImage)
+                    . // only image formats
+                    collect(Collectors.toList());
+        } catch (IOException e) {
+            LOGGER.log(
+                    Level.SEVERE,
+                    "Unable to fetch available formats for cascaded layer " + layerInfo.getName());
+            LOGGER.log(Level.SEVERE, e.getMessage());
+        }
+        return Collections.EMPTY_LIST;
+    }
+
+    @Override
+    public String getPrefferedFormat() {
+        return this.prefferedFormat;
+    }
+
+    @Override
+    public void setPrefferedFormat(String prefferedFormat) {
+        this.prefferedFormat = prefferedFormat;
+    }
+
+    @Override
+    public boolean isSelectedRemoteFormat(String format) {
+        if (prefferedFormat.equalsIgnoreCase(format)) return true;
```

Comment:
So at this stage we are 100% sure that preferred format is valid right?

Reply:
yes don't look further if the default format matches before checking the complete list

Reply:
kk